src/autowiring/test/AutoConstructTest.cpp

```diff
@@ -102,3 +102,46 @@ TEST_F(AutoConstructTest, FactoryNewPrivateCtor) {
   ASSERT_NE(nullptr, mpcc.get()) << "Null not expected as a return type from a factory new construction";
   ASSERT_EQ(1002, mpcc->ival) << "Correct ctor was not invoked on a type with a private ctor";
 }
+
+namespace {
+  class MyPrivateCtorStringClass :
+    public CoreObject
+  {
+    MyPrivateCtorStringClass(void) :
+      istr("default_string")
+    {}
+    MyPrivateCtorStringClass(const char*) :
+      istr(istr)
+    {}
+
+  public:
+    const char* istr;
+
+    static MyPrivateCtorStringClass* New(const char* str) {
+      return new MyPrivateCtorStringClass{ str };
+    }
+  };
+}
+
+static_assert(
+  autowiring::has_static_new<MyPrivateCtorStringClass, const char*>::value,
+  "Failed to find factory new on a type that carries it"
+  );
+static_assert(
+  autowiring::select_strategy<MyPrivateCtorStringClass, const char*>::value == autowiring::construction_strategy::factory_new,
+  "Construction strategy incorrectly inferred"
+  );
+static_assert(
+  !std::is_constructible<MyPrivateCtorStringClass>::value,
+  "Type reported as being constructable when it was not"
+  );
+static_assert(
+  !autowiring::has_simple_constructor<MyPrivateCtorStringClass>::value,
+  "Simple constructor detected when said constructor should have been private"
+  );
+
+TEST_F(AutoConstructTest, FactoryNewPrivateStringCtor) {
+  AutoConstruct<MyPrivateCtorStringClass> mpcc{ "a new string" };
```

Comment:
The problem with Visual Studio 2015 appears to be with converting from `const char(&)[13]` to `const char*`. There may be a problem with Visual Studio 2015, but that isn't a reason to remove the perfect forwarding. This is fixed in Visual Studio 2017, BTW. If you use cast the `"a new string"` to a `const char*`, the perfect forwarding works in Visual Studio 2015 as well.